### PR TITLE
[YUNIKORN-1364] Pass scheduler configuration from shim to core

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?
Updates the scheduler interface to support passing the core configuration directly from the shim, eliminating the need to watch configuration files directly. `RegisterResourceManagerRequest` now takes the config directly for initial population. Additionally, `UpdateConfiguration` has been removed from the SchedulerAPI since the shim is now responsible for config updates, not the core. `UpdateConfigurationRequest` / `UpdateConfigurationResponse` have been re-purposed as the carrier types for the shim to core `UpdateConfiguration` method, which now passes the updated configuration itself.

### What type of PR is it?
* [ ] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [x] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1364

### How should this be tested?
Updated unit tests downstream.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -233,6 +230,9 @@ message RegisterResourceManagerRequest {
 
   // Pass the build information of k8shim to core.
   map<string, string> buildInfo = 4;
+
+  // Pass the serialized configuration for this policyGroup to core.
+  bytes config = 5;
```
Comment (wilfred-s) [inline scheduler-interface-spec.md:235]: The yaml that we use for the config should never contain anything that cannot be represented as a normal UTF8 character. For compatibility we should use a simple `string` type. We do not want to support arbitrary binary values in the message.

File: scheduler-interface-spec.md [hunk 2]
```diff
@@ -774,19 +774,14 @@ message UpdateContainerSchedulingStateRequest {
...
 
-    // the old configuration what was changed
-    string oldConfig = 2;
+    // PolicyGroup to update
+    string policyGroup = 2;
 
-    // reason in case of failure
-    string reason = 3;
+    // New configuration to update
+    bytes config = 3;
```
Comment (wilfred-s) [inline scheduler-interface-spec.md:784]: Can we do this backward compatible and use `reserved` field number and name to indicate that one has been used. Start the new field count from 2.
I know this is overhead but lets keep it clean:
```
  reserved 1;
  reserved "configs";
```

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: scheduler-interface-spec.md (hunk 1 above)
Comment (craigcondit) [reply scheduler-interface-spec.md]: Updated.

File: scheduler-interface-spec.md (hunk 2 above)
Comment (craigcondit) [reply scheduler-interface-spec.md]: Updated.

**Review by pbacsko — COMMENTED** [review]
+1 LGTM

**Review by pbacsko — APPROVED** [review]
+1 LGTM

**Pull request closed without merging**